--- florence-0.4.6/configure.ac				2010-01-12 08:33:39.000000000 +0100
+++ florence-0.4.6/configure.ac.libm_libx11		2010-03-23 21:31:49.000000000 +0100
@@ -79,6 +79,12 @@
 	PKG_CHECK_MODULES([AT_SPI], [cspi-1.0], AC_DEFINE([ENABLE_AT_SPI], [], [at-spi enabled.]), AC_MSG_ERROR(Could not configure at-spi. Please either install at-spi or disable it: --without-at-spi configure option))
 fi
 
+# Math library
+AC_CHECK_LIB([m], [tan], AC_DEFINE(HAVE_LIBM, 1, Math lib) LIBM="-lm", AC_MSG_ERROR(Missing working libm math lib))
+
+# X11 library
+PKG_CHECK_MODULES([X11], [x11])
+
 AC_SUBST(GNOME_DOCS_UTILS_CFLAGS)
 AC_SUBST(GNOME_DOCS_UTILS_LIBS)
 AC_SUBST(LIBNOTIFY_CFLAGS)
@@ -91,6 +97,9 @@
 AC_SUBST(LIBGNOME_LIBS)
 AC_SUBST(DEPS_CFLAGS)
 AC_SUBST(DEPS_LIBS)
+AC_SUBST(X11_CFLAGS)
+AC_SUBST(X11_LIBS)
+AC_SUBST(LIBM)
 
 # Checks for header files.
 AC_FUNC_ALLOCA
--- florence-0.4.6/src/Makefile.am			2010-01-05 18:30:20.000000000 +0100
+++ florence-0.4.6/src/Makefile.am.libm_libx11		2010-03-23 20:50:24.000000000 +0100
@@ -7,7 +7,7 @@
 florence_CPPFLAGS = -DICONDIR="\"$(ICONDIR)\""\
    -DDATADIR="\"$(datadir)/florence\"" $(DEPS_CFLAGS)\
    $(LIBGNOME_CFLAGS) $(LIBNOTIFY_CFLAGS) $(XTST_CFLAGS) $(AT_SPI_CFLAGS) $(INCLUDES)
-florence_LDADD = $(DEPS_LIBS) $(LIBGNOME_LIBS) $(LIBNOTIFY_LIBS) $(XTST_LIBS) $(AT_SPI_LIBS)
+florence_LDADD = $(DEPS_LIBS) $(LIBM) $(X11_LIBS) $(LIBGNOME_LIBS) $(LIBNOTIFY_LIBS) $(XTST_LIBS) $(AT_SPI_LIBS)
 
 EXTRA_DIST = florence.h keyboard.h key.h layoutreader.h settings.h settings-window.h\
              status.h style.h system.h tools.h trace.h trayicon.h view.h
